jsf/subsystem/src/main/java/org/jboss/as/jsf/deployment/JSFPhaseListenerExtension.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat Inc., and individual contributors as indicated
+ * by the @authors tag. See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.jsf.deployment;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.BeforeBeanDiscovery;
+import javax.enterprise.inject.spi.Extension;
+import javax.faces.event.PhaseListener;
+
+/**
+ * Workaround for WFLY-6617 which makes {@link PhaseListener} suitable for @EJB injections
+ *
+ * @author Dmitrii Tikhomirov dtikhomi@redhat.com
+ */
+public class JSFPhaseListenerExtension implements Extension {
+
+    public void beforeBeanDiscovery(@Observes BeforeBeanDiscovery bbd, BeanManager bm) {
+        AnnotatedType<PhaseListener> requestedContextBean = bm.createAnnotatedType(PhaseListener.class);
```

Comment:
@treblereel Were you able to find out why injection using ```@Inject``` instead of ```@EJB``` works? I'm just wondering if this extension is actually needed or if there's something else wrong / missing that's preventing ```@EJB``` injection from working. It might be good to take a look at ```org.jboss.as.weld.services.bootstrap.WeldEjbInjectionServices``` as that's the class that's responsible for ```@EJB``` injection.

Reply:
@fjuma Hmm, i cant remember, let me take a look ...

Reply:
This does not look correct (and it also won't fix the issue for non CDI deployments). I think that these classes need to be turned into WebComponent's like we do in org.jboss.as.jsf.deployment.JSFManagedBeanProcessor.